src/composables/useGetParticipants.js

```diff
@@ -0,0 +1,135 @@
+ *
+ * @license AGPL-3.0-or-later
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+import debounce from 'debounce'
+import { ref, onMounted, nextTick, computed } from 'vue'
+
+import { subscribe, unsubscribe } from '@nextcloud/event-bus'
+
+import { useIsInCall } from './useIsInCall.js'
+import { useStore } from './useStore.js'
+import { CONVERSATION } from '../constants.js'
+import { EventBus } from '../services/EventBus.js'
+
+/**
+ * @param {boolean} isActive whether the participants tab is active
+ */
+export function useGetParticipants(isActive = true) {
+
+	// Encapsulation
+	const store = useStore()
+	const token = computed(() => store.getters.getToken())
+	const conversation = computed(() => store.getters.conversation(token.value))
+	const isInCall = useIsInCall()
+	const oneToOneConversation = computed(() => conversation.value.type === CONVERSATION.TYPE.ONE_TO_ONE)
+
+	// External use
+	const participantsInitialised = ref(false)
+	const fetchingParticipants = ref(false)
+	const pendingChanges = ref(false)
+	let debounceFastUpdateParticipants = ref(null)
+	let debounceSlowUpdateParticipants = ref(null)
+
+	/**
+	 * Initialise the get participants functionality
+	 *
+	 */
+	function initialiseGetParticipants() {
+		EventBus.$on('route-change', onRouteChange)
+		EventBus.$on('joined-conversation', onJoinedConversation)
+
+		// FIXME this works only temporary until signaling is fixed to be only on the calls
+		// Then we have to search for another solution. Maybe the room list which we update
+		// periodically gets a hash of all online sessions?
+		EventBus.$on('signaling-participant-list-changed', debounceUpdateParticipants)
+		subscribe('guest-promoted', onJoinedConversation)
+	}
+
+	/**
+	 * Stop the get participants functionality
+	 *
+	 */
+	function stopGetParticipants() {
+		EventBus.$off('route-change', onRouteChange)
+		EventBus.$off('joined-conversation', onJoinedConversation)
+		EventBus.$off('signaling-participant-list-changed', debounceUpdateParticipants)
+		unsubscribe('guest-promoted', onJoinedConversation)
+	}
+
+	const onRouteChange = () => {
+		participantsInitialised.value = store.getters.participantsList(token.value).length > 1
+	}
+
+	const onJoinedConversation = () => {
+		if (oneToOneConversation.value) {
+			cancelableGetParticipants()
+		}
+		nextTick(debounceUpdateParticipants())
+	}
```

Comment:
```suggestion
		} else {
			nextTick(debounceUpdateParticipants)
		}
	}
```
Without `else` you're fetching it twice, and not sure we need parenthesis here for a callback
Apart from that good, conversation should be already joined by the moment

Reply:
`debounceUpdateParticipants` will be called after 3sec the earliest, my intention is to add the initial request at 0sec and keep the debounce update

Reply:
But why do we need to fetch two participants one more time after 3 seconds?

Reply:
To keep a consistent interval? debounce is on each 3sec.